src/java.base/share/classes/java/lang/Byte.java

```diff
@@ -118,13 +118,15 @@ private ByteCache() {}
 
             // Load and use the archived cache if it exists
             CDS.initializeFromArchive(ByteCache.class);
-            if (archivedCache == null || archivedCache.length != size) {
+            if (archivedCache == null) {
                 Byte[] c = new Byte[size];
                 byte value = (byte)-128;
                 for(int i = 0; i < size; i++) {
                     c[i] = new Byte(value++);
                 }
                 archivedCache = c;
+            } else if (archivedCache.length != size) {
```

Comment:
The `else` case for the non-int boxed type should never happen. Add `assert archivedCache.length == size;` in the if case instead?

Reply:
Yes, it should never happen. I wanted to have a clean failure if it _does happen_, without relying on system assertions being enabled.

Reply:
Anyway, I agree that asserts should probably capture this better. We would capture this bug with `-esa` testing.